src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,19 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
```

Comment:
That's fine, one thing that might worth changing now is to pass all values of the fields (a new parameter `List<String> values`) as that can be used by the generators to return one of the provided values (other than default value).

Also, the method `getValues` could be removed (i.e. the logic moved to `getDefaultTextValue`), since we will be passing all fields to the value generators.